probe-rs/src/architecture/arm/sequences/stm32_armv6.rs

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    impl EnrG0 {
+        const ADDRESS: u64 = 0x3c;
+
+        /// Read the enable register from memory.
+        pub fn read(memory: &mut dyn ArmProbe) -> Result<Self, ArmError> {
+            let contents = memory.read_word_32(RCC + Self::ADDRESS)?;
+            Ok(Self(contents))
+        }
+
+        /// Write the enable register to memory.
+        pub fn write(&mut self, memory: &mut dyn ArmProbe) -> Result<(), ArmError> {
+            memory.write_word_32(RCC + Self::ADDRESS, self.0)
+        }
+    }
+}
+
+mod dbgmcu {
+    use crate::architecture::arm::{memory::adi_v5_memory_interface::ArmProbe, ArmError};
+    use bitfield::bitfield;
+
+    /// The base address of the DBGMCU component
+    const DBGMCU: u64 = 0x40015800;
+
+    bitfield! {
+        /// The control register (CR) of the DBGMCU. This register is described in "RM0360: STM32F0
+        /// family reference manual" section 26.9.3.
+        pub struct Control(u32);
+        impl Debug;
+
+        pub u8, dbg_standby, enable_standby_debug: 2;
+        pub u8, dbg_stop, enable_stop_debug: 1;
+    }
+
+    impl Control {
+        /// The offset of the Control register in the DBGMCU block.
+        const ADDRESS: u64 = 0x04;
+
+        /// Read the control register from memory.
+        pub fn read(memory: &mut dyn ArmProbe) -> Result<Self, ArmError> {
+            let contents = memory.read_word_32(DBGMCU + Self::ADDRESS)?;
+            Ok(Self(contents))
+        }
+
+        /// Write the control register to memory.
+        pub fn write(&mut self, memory: &mut dyn ArmProbe) -> Result<(), ArmError> {
+            memory.write_word_32(DBGMCU + Self::ADDRESS, self.0)
+        }
+    }
+}
+
+impl ArmDebugSequence for Stm32Armv6 {
+    fn debug_device_unlock(
+        &self,
+        interface: &mut dyn ArmProbeInterface,
+        default_ap: MemoryAp,
+        _permissions: &crate::Permissions,
+    ) -> Result<(), ArmError> {
+        let mut memory = interface.memory_interface(default_ap)?;
+
+        match self.family {
+            Stm32Armv6Family::F0 => {
+                let mut enr = rcc::EnrF0::read(&mut *memory)?;
+                enr.enable_dbg(true);
+                enr.write(&mut *memory)?;
+            }
+            Stm32Armv6Family::L0 => {
+                let mut enr = rcc::EnrL0::read(&mut *memory)?;
+                enr.enable_dbg(true);
+                enr.write(&mut *memory)?;
+            }
+            Stm32Armv6Family::G0 => {
+                let mut enr = rcc::EnrG0::read(&mut *memory)?;
+                enr.enable_dbg(true);
+                enr.write(&mut *memory)?;
+            }
+        }
+
+        let mut cr = dbgmcu::Control::read(&mut *memory)?;
+        cr.enable_standby_debug(true);
+        cr.enable_stop_debug(true);
+        cr.write(&mut *memory)?;
+
+        Ok(())
+    }
+
+    fn debug_core_stop(&self, interface: &mut dyn ArmProbeInterface) -> Result<(), ArmError> {
```

Comment:
Can we call this something else than core maybe? It kind of makes me think of an actual MCU core and made me think twice. Maybe `debug_module_stop`? Not sure.